libs/utils/utils.js

```diff
@@ -1059,10 +1104,14 @@ async function checkForPageMods() {
   const { init } = await import('../features/personalization/personalization.js');
```

Comment:
Let's parallelize those two calls 
![image](https://github.com/user-attachments/assets/c17799ee-73d7-4f6a-81c8-af769e3e4000)

By adding loadLink(`${getConfig().base}/utils/federated.js`, { rel: 'preload', as: 'script', crossorigin: 'anonymous' }); just before personalization on [L1058](https://github.com/adobecom/milo/pull/3258/files#diff-28218f239d8d9c4d41e1924854440b50964619e56058fdd0ae243ff5ff6c616dL1058)

Bonus points, if we actually only load federal if it's really needed within personalization. This might involve some refactoring and might be harder to do though.

Reply:
#2970 does a refactoring similar to what you're talking about, though `federated.js` was added fairly recently. I was planning to redo my work on that PR around this. So for now we can do the preload and I can conditionally load `federal.js` in a future PR.